File: data-structures/radix-sort.js
function radixSort(nums = []) {
  let high = 0;
  nums.forEach((el) => {
    if (el.length > high) high = el.length;
  });
  for (let i = 0; i < nums.length; i++) {
    while (nums[i].length < high) {
      nums[i] = "0" + nums[i];
    }
  }
  for (let i = high - 1; i >= 0; i--) {
    const subs = Array.from({ length: 10 }, () => []);
    for (let j = 0; j < nums.length; j++) {
      const pos = parseInt(nums[j].charAt(i));
      subs[pos].push(nums[j]);
    }
    nums = subs.flatMap((el) => el);
  }

  return nums;
}

console.log(
  radixSort(["31", "7", "14", "144", "111", "13", "1878", "43", "102"])
);

File: data-structures/radix_sort.py
def radix_sort(lista):
  n = 0
  for e in lista:
    if len(e) > n:
      n = len(e)
  
  for i in range(0, len(lista)):
    while len(lista[i]) < n:
      lista[i] = "0" + lista[i]

  for j in range(n-1,-1, -1):
    groups = [ [] for j in range(10)]
    for i in range(len(lista)):
      pos = int(lista[i][j])
      groups[pos].append(lista[i])
    
    lista = [ el for group in groups for el in group ]
  
  return lista

print(radix_sort(["100","4","45"]))